### PR TITLE
Add upgrade handler. Fix lint errors. Update changelog version.

### PR DESCRIPTION
**Pull request by derekadams**
## Description

Minor fixes needed for the `v1.11.1-rc1` release including:
* Add `mango` upgrade handler
* Fix minor lint errors
* Update changelog with incremented version info

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by nullpointer0x00 — DISMISSED** [review]

**Review by channa-figure — DISMISSED** [review]

**Review by Taztingo — APPROVED** [review]

**Comment by dwedul-figure** [comment]
This PR needs to go to `main` and then get cherry-picked into `release/v1.11.x`.

I don't want another situation where we have stuff in a releasex branch that isn't in main because we kept doing it this way and never updating main.

`main` first, then any release branches.

**Review by dwedul-figure — CHANGES_REQUESTED** [review]
This PR needs to go to `main` and then get cherry-picked into `release/v1.11.x`.

I don't want another situation where we have stuff in a releasex branch that isn't in main because we kept doing it this way and never updating main.

`main` first, then any release branches.

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ---
 
-## [v1.11.0](https://github.com/provenance-io/provenance/releases/tag/v1.11.0) - 2022-06-13
+## [v1.11.1-rc1](https://github.com/provenance-io/provenance/releases/tag/v1.11.1-rc1) - 2022-06-14
```
Comment (dwedul-figure) [inline CHANGELOG.md:42]: We either need to completely remove the `v1.11.0` tag or create another section for `v1.11.1` with changes that occurred between `v1.11.0` and `v1.11.1`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -37,6 +37,16 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
 
+---
+
+## [v1.11.1-rc1](https://github.com/provenance-io/provenance/releases/tag/v1.11.1-rc1) - 2022-06-14
+
+### Bug Fixes
+
+* Add `mango` upgrade handler.
+
+## [v1.11.0](https://github.com/provenance-io/provenance/releases/tag/v1.11.0) - 2022-06-13
```
Comment (dwedul-figure) [inline CHANGELOG.md:48]: Nit:
```suggestion
* Add `mango` upgrade handler.

---

## [v1.11.0](https://github.com/provenance-io/provenance/releases/tag/v1.11.0) - 2022-06-13
```

**Review by dwedul-figure — APPROVED** [review]

**Review by dwedul-figure — APPROVED** [review]

**Review by channa-figure — APPROVED** [review]

**Pull request merged**